aten/src/ATen/native/cuda/PowKernel.cu

```diff
@@ -103,9 +148,20 @@ template<typename Base_type, typename Exp_type>
 void pow_tensor_scalar_kernel_impl(TensorIteratorBase& iter,
                                                  Exp_type exp) {
   const auto d_exp = static_cast<double>(exp);
+  gpu_kernel(iter, [=]GPU_LAMBDA(Base_type base) ->Base_type {
+    if (d_exp == 2) {
+      return base * base;
+    } else if (d_exp == 3) {
+      return base * base * base;
+    } else if (d_exp == -2) {
+      return 1.0 / (base * base);
+    } else {
+      return pow_(base, exp);
+    }
+  });
```

Comment:
This would be much slower, wouldn't it, as `d_exp` have to be check for every iteration, wouldn't it? (for example in previous implementation pow(x, 3) kernel were 2 PTX instruction long, now it's 100+ and contains multiple branches)

Reply:
I'll restore separate kernels back, given that compile time is mostly not spent here anyway.